sdk/clientcore/core/src/main/java/io/clientcore/core/http/RestProxy.java

```diff
@@ -68,24 +70,30 @@ public Object invoke(Object proxy, final Method method, Object[] args) {
      * @param <A> the type of the Swagger interface
      * @return a proxy implementation of the provided Swagger interface
      */
+    @SuppressWarnings("unchecked")
     public static <A> A create(Class<A> swaggerInterface, HttpPipeline httpPipeline) {
-        return create(swaggerInterface, httpPipeline, RestProxyUtils.createDefaultSerializer());
+        final SwaggerInterfaceParser interfaceParser = SwaggerInterfaceParser.getInstance(swaggerInterface);
+        final RestProxy restProxy
+            = new RestProxy(httpPipeline, interfaceParser, new JsonSerializer(), new XmlSerializer());
+
+        return (A) Proxy.newProxyInstance(swaggerInterface.getClassLoader(), new Class<?>[] { swaggerInterface },
+            restProxy);
     }
 
     /**
      * Create a proxy implementation of the provided Swagger interface.
      *
      * @param swaggerInterface the Swagger interface to provide a proxy implementation for
-     * @param httpPipeline the HttpPipelinePolicy and HttpClient pipline that will be used to send Http requests
-     * @param serializer the serializer that will be used to convert POJOs to and from request and response bodies
+     * @param httpPipeline the HttpPipelinePolicy and HttpClient pipeline that will be used to send Http requests
+     * @param serializers the serializers that will be used to convert POJOs to and from request and response bodies
      * @param <A> the type of the Swagger interface.
-     *
      * @return a proxy implementation of the provided Swagger interface
+     * @throws IllegalArgumentException If {@code serializers} is null or empty.
      */
     @SuppressWarnings("unchecked")
-    public static <A> A create(Class<A> swaggerInterface, HttpPipeline httpPipeline, ObjectSerializer serializer) {
+    public static <A> A create(Class<A> swaggerInterface, HttpPipeline httpPipeline, ObjectSerializer... serializers) {
```

Comment:
Would it help if we use the CompositeSerializer here directly ?
Specifically, for storage use cases where we want to support both JSON and XML formats.

Reply:
For now, I'd keep `CompositeSerializer` as implementation and handle creating it from the `ObjectSerializer`s passed here. Using `RestProxy.create(Class<A>, HttpPipeline)` will be created with JSON and XML support, which should be sufficient for Storage.